src/stores/ResourceConfig.tsx

```diff
@@ -0,0 +1,655 @@
+                        removedCollections
+                    );
+                    state.resourceConfig = newResourceConfig;
+
+                    // If previous state had no collections set to first
+                    // If selected item is removed set to first.
+                    // If adding new ones set to last
+                    if (
+                        (state.collections && state.collections.length === 0) ||
+                        (state.currentCollection &&
+                            !has(state.resourceConfig, state.currentCollection))
+                    ) {
+                        state.currentCollection = newResourceKeyList[0];
+                    } else {
+                        state.currentCollection =
+                            newResourceKeyList[newResourceKeyList.length - 1];
+                    }
+
+                    // Update the collections with the new array
+                    state.collections = newResourceKeyList;
+
+                    // See if the recently updated configs have errors
+                    populateResourceConfigErrors(newResourceConfig, state);
+                }
+            }),
+            false,
+            'Resource Config Changed'
+        );
+    },
+
+    setResourceSchema: (val) => {
+        set(
+            produce((state: ResourceConfigState) => {
+                state.resourceSchema = val;
+            }),
+            false,
+            'Resource Schema Set'
+        );
+    },
+
+    stateChanged: () => {
+        const { resourceConfig } = get();
+        const { resourceConfig: initialResourceConfig } = getInitialStateData();
+
+        return !isEqual(resourceConfig.data, initialResourceConfig.data);
+    },
+
+    resetState: () => {
+        set(getInitialStateData(), false, 'Resource Config State Reset');
+    },
+});
+
+export const createHydratedResourceConfigStore = (
+    key: ResourceConfigStoreNames,
+    workflow: EntityWorkflow
+) => {
+    return create<ResourceConfigState>()(
+        devtools((set, get) => {
+            const coreState = getInitialState(set, get);
+
+            hydrateState(set, get, workflow).then(
+                () => {
+                    set(
+                        produce((state: ResourceConfigState) => {
+                            state.hydrated = true;
+                        }),
+                        false,
+                        'Resource Config State Hydrated'
+                    );
+                },
+                () => {
+                    set(
+                        produce((state: ResourceConfigState) => {
+                            const error: HydrationError = {
+                                message:
+                                    'A problem was encountered when hydrating the resource configuration store.',
+                                details:
+                                    'This error occurred in the top-level create function.',
+                            };
```

Comment:
Doesn't look like these are displayed - so I think they should not look like they are formatted for a user as they do not really add much value.

Reply:
_Pending PR draft reference._